src/dbnode/client/session.go

```diff
@@ -1300,6 +1311,11 @@ func (s *session) writeAttemptWithRLock(
 			return nil, 0, 0, err
 		}
 	}
+	nsID := s.cloneFinalizable(namespace)
+	tsID := s.cloneFinalizable(id)
+	clonedAnnotation := s.pools.checkedBytes.Get(len(annotation))
+	clonedAnnotation.IncRef()
+	clonedAnnotation.AppendAll(annotation)
```

Comment:
We should only be doing this if `len(annotation) > 0` (and then finalize it conditionally).

Reply:
https://github.com/m3db/m3/pull/3605/commits/6fc2a57b5ae8a30b318862cc127c76e72e14dc62